src/include/utils/fixed_min_heap.h

```diff
@@ -175,7 +196,12 @@ class fixed_min_pair_heap : public std::vector<std::tuple<T, U>> {
       }
 
       // Insert, since there is room
-      Base::emplace_back(x, y);
+      if constexpr (std::tuple_size_v<Tuple> == 2) {
+        Base::emplace_back(x, y);
+      } else {
+        Base::emplace_back(x, y, z);
```

Comment:
Do we need checks for the case that we insert triples in heaps initialised for pairs or other such corner cases. Also add some tests about this.